Adapt upgrade guide for katello

jlsherrill opened the pull request:
Cherry-pick into:

* [ ] Foreman 2.3 (Satellite 6.9)
* [ ] Foreman 2.1 (Satellite 6.8)

<!---
Thank you for contributing to Foreman documentation. Make sure to read README
for the documentation standards. Set cherry-pick github label to mark this
contribution for cherry picking and check which version do you need with [x].
-->

jlsherrill commented:
This is a work in progress for the upgrade docs, opening for early feedback

jlsherrill commented on guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:
```diff
@@ -37,7 +41,14 @@ Upgrading {SmartProxy} takes approximately 10 - 30 minutes.
-* Back up your {ProjectServer} and all {SmartProxyServer}s. For more information, see {AdministeringDocURL}backing-up-satellite-server-and-capsule-server[Backing Up {ProjectServer} and {SmartProxyServer}] in the _Administering {ProjectName} {ProductVersionPrevious}_ guide.
+ifeval::["{ProjectVersionPrevious}" == "2.3"]
+* Back up your {ProjectServer} and all {SmartProxyServer}s. For more information, see {SiteURL}/{ProjectVersionPrevious}/Administering_Red_Hat_Satellite/index-foreman.html#backing-up-satellite-server-and-capsule-server
+endif::[]
```
i guess this isn't actually  needed since this guide won't exist for 2.3.  TODO: DELETE THIS

lzap commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:
```diff
@@ -13,7 +13,8 @@ If you customize configuration files, manually or using a tool such as Hiera, th
 +
-For more information about backups, see https://access.redhat.com/documentation/en-us/red_hat_satellite/{ProductVersionPrevious}/html/administering_red_hat_satellite/backing-up-satellite-server-and-capsule-server[Backing Up {ProjectServer} and {SmartProxyServer}] in the _Administering {ProjectName} {ProductVersionPrevious}_ guide.
+##TODO fix this
+For more information about backups, see https://access.redhat.com/documentation/en-us/red_hat_satellite/{ProjectVersionPrevious}/html/administering_red_hat_satellite/backing-up-satellite-server-and-capsule-server[Backing Up {ProjectServer} and {SmartProxyServer}] in the _Administering {ProjectName} {ProjectVersionPrevious}_ guide.
```
TODO reminder.

melcorr replied:
What do you mean by fix this? Link to the this doc ? https://docs.theforeman.org/nightly/Administering_Red_Hat_Satellite/index-foreman-el.html#backing-up-satellite-server-and-capsule-server ? What else is needed?

jlsherrill replied:
yep, adjusting it to the new url

lzap commented on guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:
```diff
@@ -37,7 +41,14 @@ Upgrading {SmartProxy} takes approximately 10 - 30 minutes.
+
+ifeval::["{ProjectVersionPrevious}" != "2.3"]
+* Back up your {ProjectServer} and all {SmartProxyServer}s. For more information, see {SiteURL}/{ProjectVersionPrevious}/Administering_Red_Hat_Satellite/index-foreman-el.html#backing-up-satellite-server-and-capsule-server
+endif::[]
```
Nice solution :)

lzap commented:
This looks good to me, someone from Katello team should probably review tho.

melcorr commented on guides/doc-Upgrading_and_Updating/topics/cloning_satellite_server.adoc:
```diff
@@ -196,7 +196,7 @@ After you install the `satellite-clone` tool, you can adjust any configuration t
 . Follow the instructions in the `/usr/share/satellite-clone/logs/reassociate_capsules.txt` file to restore the associations between {SmartProxies} and their lifecycle environments.
-. Update your network configuration, for example, DNS, to match the target server’s IP address with its new host name. The `satellite-clone` tool changes the hostname to the source server's hostname. If you want to change the hostname to something different, you can use the `satellite-change-hostname` tool. For more information, see {AdministeringDocURL}sect-Red_Hat_Satellite-Administering_Red_Ha ...
-. If the source server uses the `virt-who` daemon, install and configure it on the target server. Copy all the `virt-who` configuration files in the `/etc/virt-who.d/` directory from the source server to the same directory on the target server.  For more information, see https://access.redhat.com/documentation/en-us/red_hat_satellite/{ProductVersion}/html/configuring_virtual_machine_subscriptions ...
+. Update your network configuration, for example, DNS, to match the target server’s IP address with its new host name. The `satellite-clone` tool changes the hostname to the source server's hostname. If you want to change the hostname to something different, you can use the `satellite-change-hostname` tool. For more information, see https://access.redhat.com/documentation/en-us/red_hat_satellite/ ...
```
```suggestion
. Update your network configuration, for example, DNS, to match the target server’s IP address with its new host name. The `satellite-clone` tool changes the hostname to the source server's hostname. If you want to change the hostname to something different, you can use the `satellite-change-hostname` tool. For more information, see {AdministeringDocURL}sect-Red_Hat_Satellite-Administering_Red_Hat_Satellite-Renaming_a_Server[Renaming a {Project} or {SmartProxyServer}] in _Administrating {ProjectName}_.
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/cloning_satellite_server.adoc:
```diff
@@ -196,7 +196,7 @@ After you install the `satellite-clone` tool, you can adjust any configuration t
-. If the source server uses the `virt-who` daemon, install and configure it on the target server. Copy all the `virt-who` configuration files in the `/etc/virt-who.d/` directory from the source server to the same directory on the target server.  For more information, see https://access.redhat.com/documentation/en-us/red_hat_satellite/{ProductVersion}/html/configuring_virtual_machine_subscriptions ...
+. Update your network configuration, for example, DNS, to match the target server’s IP address with its new host name. The `satellite-clone` tool changes the hostname to the source server's hostname. If you want to change the hostname to something different, you can use the `satellite-change-hostname` tool. For more information, see https://access.redhat.com/documentation/en-us/red_hat_satellite/ ...
+. If the source server uses the `virt-who` daemon, install and configure it on the target server. Copy all the `virt-who` configuration files in the `/etc/virt-who.d/` directory from the source server to the same directory on the target server.  For more information, see {BaseURL}configuring_virtual_machine_subscriptions_in_red_hat_satellite/index[_Configuring Virtual Machine Subscriptions in {Pr ...
 
```
This will fail the link validator as we don't have an upstream version of this link and the BASEURL will cause the linkchecker to validate the docs.theforeman.org/configuring_virtual_machine_subscriptions_in_red_hat_satellite/index.html etc etc
I removed the baseurl for that purpose.

melcorr commented on guides/common/attributes.adoc:
```diff
@@ -42,6 +43,8 @@ ifeval::["{build}" == "katello"]
 endif::[]
 ifeval::["{build}" == "satellite"]
+:ProjectVersion: {ProductVersion}
+:ProjectVersionPrevious: {ProductVersionPrevious}
```
Oh oh oh. I love this!

melcorr commented on guides/common/attributes.adoc:
```diff
@@ -130,7 +136,8 @@ ifdef::katello,foreman-el[]
 :ansible-namespace-example: `theforeman.foreman._module_name_`
 :awx: AWX
-:BaseURL: https://docs.theforeman.org/nightly/
+:SiteURL: https://docs.theforeman.org
```
Can you explain what this does that wasn't possible beforehand?

jlsherrill replied:
it made it hard to refer to the previous version of the docs, like how to backup the previous version

melcorr reviewed (commented):
All looks good. A question here or there/

ehelms commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:
```diff
@@ -68,39 +69,70 @@ If you lose connection to the command shell where the upgrade command is running
+----
+  yum update -y https://fedorapeople.org/groups/katello/releases/yum/{KatelloVersion}/katello/el7/x86_64/katello-repos-latest.rpm
+  yum update -y https://yum.theforeman.org/releases/{ProjectVersion}/el7/x86_64/foreman-release.rpm
+  yum install -y centos-release-scl-rh
```
Can drop the indention here to match up with other declarations

ehelms commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:
```diff
@@ -68,39 +69,70 @@ If you lose connection to the command shell where the upgrade command is running
++
+[options="nowrap" subs="attributes"]
+----
+foreman-installer
```
There is a macro for this, I think `{foreman-installer}`

melcorr commented:
Hey @jlsherrill - is there anything I've blocked you with here? 
@lzap  is proposing a large change with #460 that will impact your proposed attributes also. 
I'm assuming this will need to be finished soon enough?

jlsherrill commented:
Sorry, about that!  i will update this today

jlsherrill commented:
i think i've addressed all the comments, i think i'm getting 404s because i've got links referring to the '2.4' docs, but they are currently at /nightly/.  Will work on that!

jlsherrill commented:
🟢

jlsherrill commented:
I just realized i could do this! 

🔴 | 
-------|
🟡  |
🟢  |

lzap commented:
Feel free to merge what comes first, I can rebase my PR too.

I am good with this one, let's merge?

ekohl commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:
```diff
@@ -68,39 +74,70 @@ If you lose connection to the command shell where the upgrade command is running
+----
+# yum update -y https://fedorapeople.org/groups/katello/releases/yum/{KatelloVersion}/katello/el7/x86_64/katello-repos-latest.rpm
+# yum update -y https://yum.theforeman.org/releases/{ProjectVersion}/el7/x86_64/foreman-release.rpm
+# yum install -y centos-release-scl-rh
```
Install here is weird. It should already be present, so upgrade?
```suggestion
# yum update -y centos-release-scl-rh
```

Also, this is a CentOS users part. RHEL users should enable the right channel instead.

Thinking more about it, this should already be covered in the `yum update -y` just before this so:
```suggestion
```

ekohl commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:
```diff
@@ -68,39 +74,70 @@ If you lose connection to the command shell where the upgrade command is running
++
+[options="nowrap" subs="attributes"]
+----
+# yum clean all
```
I think you don't need to clean everything. Perhaps
```suggestion
# yum clean expire-cache
```

Or if that's insufficient:
```suggestion
# yum clean metadata
```

ekohl commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:
```diff
@@ -68,39 +74,70 @@ If you lose connection to the command shell where the upgrade command is running
+.For Centos 7 or Red Hat Enterprise Linux Users:
+[options="nowrap" subs="attributes"]
+----
+# yum update -y https://fedorapeople.org/groups/katello/releases/yum/{KatelloVersion}/katello/el7/x86_64/katello-repos-latest.rpm
```
I know @evgeni is working on moving this. Perhaps https://yum.theforeman.org/katello/ should be used already?

evgeni replied:
Yes please. 3.17, 3.18 and 4.0 is installable from yum.tfm today.

ekohl commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:
```diff
@@ -68,39 +74,70 @@ If you lose connection to the command shell where the upgrade command is running
+[options="nowrap" subs="attributes"]
+----
+# yum update -y https://fedorapeople.org/groups/katello/releases/yum/{KatelloVersion}/katello/el7/x86_64/katello-repos-latest.rpm
+# yum update -y https://yum.theforeman.org/releases/{ProjectVersion}/el7/x86_64/foreman-release.rpm
```
I would like to suggest to swap the order: first `foreman-release.rpm` and then update katello. Mostly for a future simplification: I think that we should make a `foreman-release-katello` in the Foreman repository. Then the installation instructions could become:
```
yum install -y https://yum.theforeman.org/releases/{ProjectVersion}/el7/x86_64/foreman-release.rpm
yum install -y foreman-release-katello centos-release-scl-rh
```
(name could also be `katello-release`, not important for the example)

The benefit is that there is no (reasonable) way for a user to install a mismatched version.

ekohl commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:
```diff
@@ -68,39 +74,70 @@ If you lose connection to the command shell where the upgrade command is running
 # rpm -qa --last | grep kernel
 ----
-
 . Optional: If a kernel update occurred since the last reboot, stop the `{foreman-maintain}` services and reboot the system:
```
I don't understand this suggestion. Why do you need to stop all services and is just `reboot` insufficient? Is that always required? If so, IMHO it's a major limitation that we must think about.

jlsherrill replied:
i agree, it doesn't make much sense.  Let me change to just 'reboot the system'.

ekohl commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:
```diff
@@ -68,39 +74,70 @@ If you lose connection to the command shell where the upgrade command is running
 # hash -d {foreman-maintain} service 2> /dev/null
 ----
-
 . Optional: If you made manual edits to DNS or DHCP configuration files, check and restore any changes required to the DNS and DHCP configuration files using the backups that you make.
```
IMHO this is unsupported and we should not document it. https://theforeman.org/plugins/katello/nightly/installation/index.html#multiple-subnets-and-domains documents how to handle this in the installer. Perhaps out of scope for this PR.

jlsherrill replied:
While i agree, i think maybe out of scope for this pr, maybe we should open an issue to incorporate https://theforeman.org/plugins/katello/nightly/installation/index.html#multiple-subnets-and-domains

ekohl commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:
```diff
@@ -68,39 +74,70 @@ If you lose connection to the command shell where the upgrade command is running
 # {foreman-maintain} service restart
 ----
-
 . If you have the OpenSCAP plug-in installed, but do not have the default OpenSCAP content available, enter the following command:
```
When would this happen? I don't know OpenSCAP but is this something seeding should take care of? If so, IMHO it can indicate a bug that we have to document this.

jlsherrill replied:
i'm deleting this, as i don't understand why its needed

ekohl commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:
```diff
@@ -33,7 +39,7 @@ For more information about backups, see https://access.redhat.com/documentation/
 ----
```
While you're at it, could you hide the above section behind a katello/satellite definition? Vanilla Foreman doesn't need `rh-postgresql12-postgresql-evr`.

ekohl commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:
```diff
@@ -13,7 +13,13 @@ If you customize configuration files, manually or using a tool such as Hiera, th
+For more information about backups, see {SiteURL}/{ProjectVersionPrevious}/administering_red_hat_satellite/index-foreman-el.html#backing-up-satellite-server-and-capsule-server[Backing Up {ProjectServer} and {SmartProxyServer}] in the _Administering {ProjectName} {ProjectVersionPrevious}_ guide.
+endif:[]
 
 . Optional: If you made manual edits to DNS or DHCP configuration in the `/etc/zones.conf` or `/etc/dhcp/dhcpd.conf` files, back up the configuration files because the installer only supports one domain or subnet, and therefore restoring changes from these backups might be required.
```
Also noted below, but personally I think we should not encourage this.

ekohl commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:
```diff
@@ -13,7 +13,13 @@ If you customize configuration files, manually or using a tool such as Hiera, th
 +
-For more information about backups, see https://access.redhat.com/documentation/en-us/red_hat_satellite/{ProductVersionPrevious}/html/administering_red_hat_satellite/backing-up-satellite-server-and-capsule-server[Backing Up {ProjectServer} and {SmartProxyServer}] in the _Administering {ProjectName} {ProductVersionPrevious}_ guide.
+ifeval::["{ProjectVersionPrevious}" == "2.3"]
+For more information about backups, see {SiteURL}/{ProjectVersion}/administering_red_hat_satellite/index-foreman-el.html#backing-up-satellite-server-and-capsule-server[Backing Up {ProjectServer} and {SmartProxyServer}] in the _Administering {ProjectName} {ProjectVersionPrevious}_ guide.
```
While the whole document is very focused on EL, we may want to create an issue to write similar instructions for Debian.

lzap commented:
Hello,

we have merged a big cleanup in `attributes.adoc` file, it is now split into several files:

* [attributes.adoc](https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes.adoc): version definitions and includes for other attribute files.
* [attributes-base.adoc](https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-base.adoc): base attributes common for all builds.
* [attributes-foreman-el.adoc](https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-foreman-el.adoc): base overrides for foreman-el build.
* [attributes-foreman-deb.adoc](https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-foreman-deb.adoc): base overrides for foreman-deb build.
* [attributes-katello.adoc](https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-katello.adoc): base overrides for katello build.
* [attributes-satellite.adoc](https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-satellite.adoc): base overrides for satellite build.

Please resolve the conflicts and update the attributes accordingly, sorry for this but we believe this will ultimately lead to cleaner attributes. I wish you nice Easter!

lzap commented:
I am about to branch off 2.4 series, what is the status please? I can hold off the branching process if this is something that should be in 2.4/4.0. Let me know, thanks.

jlsherrill commented:
Apologies, i will work on this tonight :)

jlsherrill commented:
@ekohl i think i addressed all your comments other than those you mentioned might be out of scope for this PR

jlsherrill commented:
Thanks @lzap for the patience!

lzap commented:
Can I get re-reviews please? Thank you all!

melcorr commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:
```diff
@@ -70,41 +75,65 @@ If you lose connection to the command shell where the upgrade command is running
-
 . If you make changes in the previous step, restart the `{foreman-maintain}` services.
 +
 [options="nowrap"]
```
This isn't your fault but 
```suggestion
[options="nowrap" subs="attributes"]
```

melcorr reviewed (changes requested):
Hi @jlsherrill 

Can you do a local test of this? I think that the master.adoc file might need some tweaking. The nav menu that I see only has the following sections:

    1. Upgrade Overview
        1.1. Prerequisites
        1.2. Post-Upgrade Tasks
        1.3. Reclaiming MongoDB Space
        1.4. Reclaiming PostgreSQL Space
        1.5. Updating Satellite Server, Capsule Server, and Content Hosts

I cannot find a lot of the content that you've edited here. I know you've suppressed some Satellite-only content but this seems like more than this.

jlsherrill commented on guides/doc-Upgrading_and_Updating/master.adoc:
```diff
@@ -40,6 +42,7 @@ endif::[]
 
 // Synchronizing the New Repositories
 include::topics/synchronizing_the_new_repositories.adoc[leveloffset=+2]
+endif::[]
```
i think this was a rebase error, good catch!

jlsherrill commented:
@melcorr i did not test it after rebasing, apparently i should have!   Fixed, as well as removed some mongo docs.  

I noticed that the smart proxy upgrade part also needs a it of work, but i propose to work on this as a followup

melcorr commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:
```diff
@@ -70,41 +69,65 @@ If you lose connection to the command shell where the upgrade command is running
+# yum clean all
+# yum -y update
+----
+. Run installer
```
```suggestion
. Run the installer
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:
```diff
@@ -47,23 +52,37 @@ If you customize configuration files, manually or use a tool such as Hiera, thes
 Because of the lengthy upgrade time, use a utility such as `screen` to suspend and reattach a communication session. You can then check the upgrade progress without staying connected to the command shell continuously. For more information about using the screen command, see link:https://access.redhat.com/articles/5247[How do I use the screen command?] article in the _Red{nbsp}Hat Knowledge{nbsp}B ...
 
-If you lose connection to the command shell where the upgrade command is running you can see the logs in `/var/log/foreman-installer/satellite.log` to check if the process completed successfully.
+If you lose connection to the command shell where the upgrade command is running you can see the logs in `{installer-log-file}` to check if the process completed successfully.
```
Is installer-log-file defined anywhere? When I build this as Katello, I get the following: 

If you lose connection to the command shell where the upgrade command is running, you can see the logged messages in the /var/log/foreman-installer/satellite.log file to check if the process completed successfully.

melcorr commented on guides/doc-Upgrading_and_Updating/topics/introduction_updating_satellite.adoc:
```diff
@@ -1,6 +1,6 @@
 [[introduction_updating_satellite]]
```
The title for this is rendering as: 

Updating Satellite Server, Capsule Server, and Content Hosts

melcorr commented on guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:
```diff
@@ -37,7 +41,8 @@ Upgrading {SmartProxy} takes approximately 10 - 30 minutes.
 * Ensure that you have sufficient storage space on your server. For more information, see {InstallingProjectDocURL}preparing-environment-for-satellite-installation[Preparing your Environment for Installation] in _Installing {ProjectServer} from a Connected Network_ and {InstallingSmartProxyDocURL}preparing-environment-for-capsule-installation[Preparing your Environment for Installation] in _Insta ...
 
-* Back up your {ProjectServer} and all {SmartProxyServer}s. For more information, see {AdministeringDocURL}backing-up-satellite-server-and-capsule-server[Backing Up {ProjectServer} and {SmartProxyServer}] in the _Administering {ProjectName} {ProductVersionPrevious}_ guide.
+* Back up your {ProjectServer} and all {SmartProxyServer}s. For more information, see {AdministeringDocURL}#backing-up-satellite-server-and-capsule-server
```
```suggestion
* Back up your {ProjectServer} and all {SmartProxyServer}s. For more information, see {AdministeringDocURL}#backing-up-satellite-server-and-capsule-server[Backing Up {ProjectServer} and {SmartProxyServer}] in the _Administering {ProjectName} {ProductVersionPrevious}_ guide.
```
Without the text in [] this just prints a hyperlink. It's good practice to recommend the previous version of the guide because the backup and restore docs might shift from version to version.

melcorr commented on guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:
```diff
@@ -47,23 +52,37 @@ If you customize configuration files, manually or use a tool such as Hiera, thes
 
-The high level steps in upgrading to {Project} {ProductVersion} are as follows.
+The high level steps in upgrading to {Project} {ProjectVersion} are as follows.
 
 . Clone your existing {ProjectServer}s. For more information, see xref:cloning_satellite_server[].
```
```suggestion
ifdef::satellite[]
. Clone your existing {ProjectServer}s. For more information, see xref:cloning_satellite_server[].
endif::[]
```

This is broken as there are no cloning instructions included in the Katello upgrade guide?

melcorr commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:
```diff
@@ -70,41 +69,65 @@ If you lose connection to the command shell where the upgrade command is running
+# yum install -y centos-release-scl-rh
+----
+. Update packages
+Clean the yum cache and update required packages
```
```suggestion
Clean the yum cache and update the required packages
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc:
```diff
@@ -2,12 +2,12 @@
 * You must upgrade {ProjectServer} before you can upgrade any {SmartProxyServer}s. Note that you can upgrade {SmartProxies} separately from {Project}. For more information, see xref:upgrading_capsules-separately-from-satellite_upgrade-guide[].
-* Ensure the {ProjectName} {SmartProxy} {ProductVersion} repository is enabled in {ProjectServer} and synchronized.
+* Ensure the {ProjectName} {SmartProxy} {ProjectVersion} repository is enabled in {ProjectServer} and synchronized.
 * Ensure that you synchronize the required repositories on {ProjectServer}. For more information, see xref:synchronizing_the_new_repositories[].
```
The xref here doesn't work because this section has been filtered out of the Katello guide. Do we need it? If so, add it in, if not, wrap it in a Satellite ifdef

melcorr reviewed (changes requested):
Hi Justin, 
A few final things!

lzap commented:
Once you merge this @melcorr please ping me so I can rebase and finalize the branching process.

melcorr commented:
@jlsherrill if you do not have time, please let me know, I'll merge then apply these suggestions myself. I know this has been a PITA

jlsherrill commented:
@melcorr if you don't mind, that would be amazing, I'm head deep in some fixes for 6.9.1 :)

melcorr commented:
@lzap  please don't branch until later when I finish this !!

melcorr commented:
@lzap i completed this in #488 - are we rushing a bit with the branching? Katello 4.0 is the first time we will be official. No one is expecting us to have 2.4. If we merge now, we have a lot of cherry-picking. What do you think?
